pkg/kv/kvserver/replica_consistency.go

```diff
@@ -51,10 +51,12 @@ import (
 // Up to 22.1, the consistency check initiator used to synchronously collect the
 // first replica's checksum before all others, so checksum collection requests
 // could arrive late if the first one was slow. Since 22.2, all requests are
-// parallel and likely arrive quickly.
+// parallel and likely arrive quickly. Thus, in 23.1 the checksum task waits a
+// short amount of time until the collection request arrives, and otherwise
+// doesn't start.
 //
-// TODO(pavelkalinnikov): Consider removing GC behaviour in 23.1+, when all the
-// incoming requests are from 22.2+ nodes (hence arrive timely).
+// We still need the delayed GC in order to help a late arriving participant to
+// learn that the other one gave up, and fail fast instead of waiting.
```

Comment:
If the timeout here is only 5 seconds, I have a slight preference for just getting rid of this. It won't happen in the typical case, and when it does, it's usually an async caller who doesn't really care if it takes 5 seconds to fail.

Reply:
Will address this in a follow-up PR, as this one is getting too big :)

Reply:
#88902